docs/site/Interceptors.md

```diff
@@ -0,0 +1,529 @@
+     return `Hello, ${name}`;
+   }
+   ```
+
+4. greet: [`convertName`, `log`]
+
+   ```ts
+   // Apply multiple interceptors. The order of `log` will be preserved as it
+   // explicitly listed at method level
+   @intercept(convertName, log)
+   async greet(name: string) {
+     return `Hello, ${name}`;
+   }
+   ```
+
+## Create your own interceptors
+
+Interceptors can be made available by LoopBack itself, extension modules, or
+applications. They can be a function that implements `Interceptor` signature or
+a binding that is resolved to an `Interceptor` function.
+
+### Interceptor functions
+
+The interceptor function has the following signature:
+
+```ts
+export interface Interceptor {
+  <T = unknown>(
+    context: InvocationContext,
+    next: () => ValueOrPromise<T>,
+  ): ValueOrPromise<T>;
+}
+```
+
+An interceptor can be synchronous (returning a value) or asynchronous (returning
+a promise). If one of the interceptors or the target method is asynchronous, the
+invocation will be asynchronous.
+
+### Invocation context
+
+The `InvocationContext` object provides access to metadata for the given
+invocation in addition to the parent `Context` that can be used to locate other
+bindings.
+
+```ts
+/**
+ * InvocationContext for method invocations
+ */
+export class InvocationContext extends Context {
+  /**
+   * Construct a new instance
+   * @param parent Parent context
+   * @param target Target class or object
+   * @param methodName Method name
+   * @param args An array of arguments
+   */
+  constructor(
+    parent: Context,
+    public readonly target: object,
+    public readonly methodName: string,
+    public readonly args: any[],
+  ) {
+    super(parent);
+  }
+}
+```
+
+It's possible for an interceptor to mutate items in the `args` array to pass in
+transformed input to downstream interceptors and the target method.
+
+### Logic around `next`
+
+An interceptor will receive the `next` parameter, which is a function to execute
+the downstream interceptors and the target method.
+
+The interceptor function is responsible for calling `next()` if it wants to
+proceed with next interceptor or the target method invocation. A typical
+interceptor implementation looks like the following:
+
+```ts
+async function intercept<T>(
+  invocationCtx: InvocationContext,
+  next: () => Promise<T>,
+) {
+  // Pre-process the request
+  try {
+    const result = await next();
+    // Post-process the response
+    return result;
+  } catch (err) {
+    // Handle errors
+    throw err;
+  }
+}
+```
+
+If `next()` is not invoked, neither downstream interceptors nor the target
+method be executed. It's valid to skip `next()` if by intention, for example, an
+interceptor can fail the invocation early due to validation errors or return a
+response from cache without invoking the target method.
+
+### Example interceptors
+
+Here are some example interceptor functions:
+
+```ts
+// A sync interceptor to log method calls
```

Comment:
Please split this long code snippet into multiple smaller ones. Rework the comment introducing each interceptor implementation into a title. For example:

```markdown
### Example interceptors
	
Here are some example interceptor functions:

#### A sync interceptor to log method calls

``ts
const logSync: Interceptor = (invocationCtx, next) => {
  // ...
};
``

#### An async interceptor to log method calls

etc.
```